src/System.Text.Json/System/Text/Json/JsonParseObject.cs

```diff
@@ -182,6 +182,19 @@ public bool TryGetValue(string propertyName, out JsonObject value)
             }
         }
 
+        public int Length
+        {
+            get
+            {
+                var record = Record;
+                if (record.Type != JsonValueType.Array)
+                {
+                    throw new NullReferenceException();
```

Comment:
This should throw InvalidOperationException, which indicates that the call could work if the object was in some other state, i.e. if the JsonObject was an array.